engine/lite/k8sDeploymentManager/src/main/scala/pl/touk/nussknacker/k8s/manager/deployment/K8sScalingOptionsDeterminer.scala

```diff
@@ -0,0 +1,76 @@
+package pl.touk.nussknacker.k8s.manager.deployment
+
+import net.ceedubs.ficus.Ficus
+import net.ceedubs.ficus.Ficus._
+import net.ceedubs.ficus.readers.ArbitraryTypeReader._
+import net.ceedubs.ficus.readers.ValueReader
+import pl.touk.nussknacker.k8s.manager.deployment.K8sScalingConfig._
+
+trait K8sScalingOptionsDeterminer {
+
+  def determine(parallelism: Int): K8sScalingOptions
+
+}
+
+case class K8sScalingOptions(replicasCount: Int, noOfTasksInReplica: Int)
+
+object K8sScalingOptionsDeterminer {
+
+  // 4 because it is quite normal number of cpus reserved for one container
```

Comment:
I wonder, probably we could connect it to configured resources somehow? For our use case - scenario per pod - 4 seems quite large number?

Reply:
> connect it to configured resources somehow
I don't know what you have in mind by "configured resources"?
If we will need some better scaling options determiner, we can always change this strategy or add new. `K8sScalingOptions` can return more info, and `determine(...)` can take more parameters. But let's talk about some concrete target that we want to achieve.
Also I don't like to shuffle logic of scaling with "low level" `k8sDeploymentConfig` - it will become too hard to understand.  
> scenario per pod - 4 seems quite large number
why? why it depends on use case? It shouldn't depend only on available resources?

Reply:
1. in the future we will probably set some default resource limits (e.g. on CPU), if they will be sth like "100m", then 4 threads is too much
2. for our target performance "10k/s per thread" 4 threads seems quite much :) 

you can ignore this comment, just wanted to discuss a bit :)

Reply:
probable you've right, but currently I don't have enough knowledge to do tuning of defaults. My thinking was - If our jobs are "fat" we should have lower number of them instead of higher. But I can be wrong - let's do tuning after performance tests